database.go

```diff
@@ -0,0 +1,1256 @@
+type CloudDatabaseAddressesDetail struct {
+	IPAddress   string `json:"ip_address"`
+	NetworkName string `json:"network_name"`
+}
+
+// CloudDatabaseAddresses contains addresses information of database.
+type CloudDatabaseAddresses struct {
+	Private []CloudDatabaseAddressesDetail `json:"private"`
+	Public  []CloudDatabaseAddressesDetail `json:"public"`
+}
+
+// CloudDatabaseDatastore contains datastore information.
+type CloudDatabaseDatastore struct {
+	Type string `json:"type"`
+	ID   string `json:"id,omitempty"`
+	Name string `json:"name,omitempty"`
+}
+
+// CloudDatabaseAction contains database instance action information.
+type CloudDatabaseAction struct {
+	Action     string `json:"action"`
+	NewSize    int    `json:"new_size,omitempty"`
+	FlavorName string `json:"flavor_name,omitempty"`
+	ResizeAll  bool   `json:"resize_all,omitempty"`
+	ActionAll  bool   `json:"action_all,omitempty"`
+}
+
+// CloudDatabaseDelete contains database delete information.
+type CloudDatabaseDelete struct {
+	PurgeBackup     bool `json:"purge_backup,omitempty"`
+	PurgeAutobackup bool `json:"purge_autobackup,omitempty"`
+}
+
+// CloudDatabaseNodeCreateDetail contains detail to create database node payload.
+type CloudDatabaseNodeCreateDetail struct {
+	ReplicaOf        string `json:"replica_of"`
+	AvailabilityZone string `json:"availability_zone"`
+	Role             string `json:"role"`
+}
+
+// CloudDatabaseNodeCreate contains wrap of database not create payload.
+type CloudDatabaseNodeCreate struct {
+	DatabaseNode CloudDatabaseNodeCreateDetail `json:"database_node"`
+}
+
+// CloudDatabaseVolume contains info of volume.
+type CloudDatabaseVolume struct {
+	Size int     `json:"size"`
+	Used float32 `json:"used"`
+}
+
+// CloudDatabaseNode contains detail of a database node.
+type CloudDatabaseNode struct {
+	ID               string                 `json:"id"`
+	Name             string                 `json:"name"`
+	Status           string                 `json:"status,omitempty"`
+	Flavor           string                 `json:"flavor,omitempty"`
+	Datastore        CloudDatabaseDatastore `json:"datastore,omitempty"`
+	RegionName       string                 `json:"region_name,omitempty"`
+	AvailabilityZone string                 `json:"availability_zone"`
+	Volume           CloudDatabaseVolume    `json:"volume,omitempty"`
+	CreatedAt        string                 `json:"created_at,omitempty"`
+	DNS              CloudDatabaseDNS       `json:"dns"`
+	Addresses        CloudDatabaseAddresses `json:"addresses"`
+	MetaData         map[string]interface{} `json:"meta_data,omitempty"`
+	Role             string                 `json:"role"`
+	InstanceID       string                 `json:"instance_id,omitempty"`
+	Description      string                 `json:"description,omitempty"`
+	Replicas         []CloudDatabaseNode    `json:"replicas,omitempty"`
+	Message          string                 `json:"message,omitempty"`
```

Comment:
Beacause this is the required field when show a node. If the value is none, It will return default "" or 0 or {} instead of nil. That will help user save it in database if need more easier.